components/delinquency_modal/index.ts

```diff
@@ -0,0 +1,44 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import {connect} from 'react-redux';
+import {bindActionCreators, Dispatch} from 'redux';
+
+import {getLicense} from 'mattermost-redux/selectors/entities/general';
+import {makeGetCategory} from 'mattermost-redux/selectors/entities/preferences';
+import {GenericAction} from 'mattermost-redux/types/actions';
+import {getCloudSubscription} from 'mattermost-redux/actions/cloud';
+import {isCurrentUserSystemAdmin} from 'mattermost-redux/selectors/entities/users';
+
+import {ModalIdentifiers, Preferences} from 'utils/constants';
+import {GlobalState} from 'types/store';
+import {closeModal, openModal} from 'actions/views/modals';
+
+import DeliquencyModalController from './delinquency_modal_controller';
+
+function mapStateToProps(state: GlobalState) {
+    const getCategory = makeGetCategory();
+    const license = getLicense(state);
+    const isCloud = license.Cloud === 'true';
+    const subscription = state.entities.cloud?.subscription;
+    const userIsAdmin = isCurrentUserSystemAdmin(state);
+
+    return {
+        isCloud,
+        subscription,
+        userIsAdmin,
+        delinquencyModalPreferencesConfirmed: getCategory(state, Preferences.DELINQUENCY_MODAL_CONFIRMED),
+    };
+}
+
+function mapDispatchToProps(dispatch: Dispatch<GenericAction>) {
+    return {
+        actions: bindActionCreators({
+            getCloudSubscription,
+            closeModal: () => closeModal(ModalIdentifiers.DELINQUENCY_MODAL_DOWNGRADE),
```

Comment:
Similarly, this isn't actually needed since the `onExited` prop provided by the `ModalController` does this for us